### PR TITLE
fix: divergence in `command_argument_count`

### LINKED ISSUE #4298: bug: `command_argument_count()` inquiry function returns incorrect output
```fortran
program main
   implicit none
   integer :: narg
   narg = command_argument_count()
   print *, "narg: ", narg
end program main
```
```console
(base) saurabh-kumar@Awadh:~/Projects/System/lfortran$ gfortran ./examples/example.f90
(base) saurabh-kumar@Awadh:~/Projects/System/lfortran$ ./a.out a.txt b.txt c.txt
 narg:            3
(base) saurabh-kumar@Awadh:~/Projects/System/lfortran$ lfortran ./examples/example.f90 -o tmp
(base) saurabh-kumar@Awadh:~/Projects/System/lfortran$ ./tmp a.txt b.txt c.txt
narg:  4
```

This bug is the cause of the `***ERROR: CMDARG: Missing command line entry` error when executing `./gsnap` with command-line arguments after compiling SNAP using LFortran.

### PR DESCRIPTION
**Pull request by Pranavchiku**
Fixes #4298.

### CODE REVIEW

File: src/libasr/codegen/asr_to_llvm.cpp [hunk 1]
```diff
@@ -9501,12 +9501,12 @@ class ASRToLLVMVisitor : public ASR::BaseVisitor<ASRToLLVMVisitor>
...
                 } else if (func_name == "command_argument_count") {
-                    llvm::Function *fn = module->getFunction("_lpython_get_argc");
+                    llvm::Function *fn = module->getFunction("_lfortran_get_argc");
                     if(!fn) {
                         llvm::FunctionType *function_type = llvm::FunctionType::get(
                             llvm::Type::getInt32Ty(context), {}, false);
                         fn = llvm::Function::Create(function_type,
-                            llvm::Function::ExternalLinkage, "_lpython_get_argc", *module);
+                            llvm::Function::ExternalLinkage, "_lfortran_get_argc", *module);
                     }
```
Comment (Pranavchiku) [inline src/libasr/codegen/asr_to_llvm.cpp:9510]: I checked at `lpython` frontend, we do not use `command_argument_count` thus, we can safely do this.

**Comment by Pranavchiku** [comment]
CI failure is because in `wasm` backend, `argc` was already set to 0 correctly. 

```
 528/1042 Test  #529: intrinsics_290 ...........................***Failed    0.15 sec
narg:  -1
ERROR STOP
```

**Comment by assem2002** [comment]
doesn't wasm count the program name? or while passing argc and argv to the c code it doesn't pass the program name in the first place?
If it's the last case, We could initialize `char** _argv` with `nullptr` and checking whether to subtract 1 or not based on checking _argv is set to `nullptr` or not. maybe that could fix that case.

**Comment by Pranavchiku** [comment]
Nice idea. I'll try that. If you have any diff in mind please feel free to suggest.

**Comment by Pranavchiku** [comment]
I am not exactly sure of how to fix the `WASM` failure probably they do not pass any executable and thus the count remains 0. I disabled `llvm_wasm` and `llvm_wasm_emcc` for now in `intrinsics_292`.

File: src/libasr/codegen/asr_to_llvm.cpp (hunk 1 above)
Comment (certik) [reply src/libasr/codegen/asr_to_llvm.cpp:9510]: We are currently not using it, but we should use command_argument_count  in LPython also.

**Review by certik — APPROVED** [review]
I think this is fine for now to get us unblocked. Eventually we should get LPython to use this intrinsic also.

**Pull request merged**